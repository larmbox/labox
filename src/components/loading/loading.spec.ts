import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { LLoading } from '../../';

test('mount component', async () => {
  expect(LLoading).toBeTruthy();

  const wrapper = mount(LLoading, {
    props: {
      id: 'id',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
  expect(wrapper.html()).toContain('loading');
});

test('register css variables', async () => {
  mount(LLoading);
  expect(
    document.getElementById('lx-theme-provider-global')!.innerText
  ).toContain('--lx-loading');
});

test('inherit attributes', async () => {
  const component = mount(LLoading, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
