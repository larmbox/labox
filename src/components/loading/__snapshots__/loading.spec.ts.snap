// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div id=\\"id\\" class=\\"loading loading-spinner loading-md loading-primary\\" aria-busy=\\"true\\" role=\\"progressbar\\">
  <div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;
